### PR TITLE
6-get-review-id-comments

### PR DESCRIPTION
**Pull request by reawilliams20**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -106,4 +106,50 @@ describe('/api/reviews/:review_id', () => {
...
+        })
+    })
+    test('404: valid id but no comments for it, responds with 404 error message', () => {
+        return request(app)
+        .get('/api/reviews/1/comments')
+        .expect(404)
+        .then((response) => {
+            const msg = response.body.msg
+            expect(msg).toBe('Not found')
+        })
```
Comment (Dutton1990) [inline __tests__/app.test.js:156]: This should be a 200, the request was correct and successful but there aren't any comments _yet_. If you viewed an article online without any comments yet it wouldn't error.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -106,4 +106,50 @@ describe('/api/reviews/:review_id', () => {
...
+            expect(comments).toBeSortedBy('created_at', { ascending: true });
+            comments.forEach((comment) => {
+                expect(comment).toEqual(
+                    expect.objectContaining({
+                        comment_id: expect.any(Number),
+                        votes: expect.any(Number),
+                        created_at: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                        review_id: expect.any(Number)
```
Comment (Dutton1990) [inline __tests__/app.test.js]: you can be more specific with the review_id, it should always be 3 even if we don't know exactly what the other data is

File: __tests__/app.test.js [hunk 3]
```diff
@@ -106,4 +106,50 @@ describe('/api/reviews/:review_id', () => {
...
+
+
+  describe("GET /api/reviews/:review_id/comments", () => {
+    test('200: responds with corresponding comments of given review_id', () => {
+        const review_id = 3;
+        return request(app)
+        .get(`/api/reviews/${review_id}/comments`)
+        .expect(200)
+        .then(({body: {comments}}) => {
+            console.log(comments)
```
Comment (Dutton1990) [inline __tests__/app.test.js]: dont forget to remove console logs!

File: __tests__/app.test.js (hunk 1 above)
Comment (Dutton1990) [reply __tests__/app.test.js:156]: You should add another test for a valid id but the review doesn't exist, so 999 for example 😄

**Review by Dutton1990 — CHANGES_REQUESTED** [review]
Just some suggestions above! 

You'll have to consider how you can check whether the review exists but simply has no comments to cater for that 200 i've suggested!

Once you've made those changes please send the PR back over to us 😄

File: __tests__/app.test.js [hunk 4]
```diff
@@ -106,4 +106,64 @@ describe('/api/reviews/:review_id', () => {
...
+
+
+  describe("GET /api/reviews/:review_id/comments", () => {
+    test('200: responds with corresponding comments of given review_id', () => {
+        const review_id = 3;
+        return request(app)
+        .get(`/api/reviews/${review_id}/comments`)
+        .expect(200)
+        .then(({body: {comments}}) => {
+            expect(comments).toBeInstanceOf(Array);
```
Comment (RichardNeat) [inline __tests__/app.test.js]: we don't need to explicitly check for an array here as we are using a forEach later on that is implicit that it must be an array

File: __tests__/app.test.js [hunk 5]
```diff
@@ -106,4 +106,64 @@ describe('/api/reviews/:review_id', () => {
...
+                )
+            })
+        })
+    })
+    test('200: valid id but no comments for it, responds with empty array', () => {
+        return request(app)
+        .get('/api/reviews/1/comments')
+        .expect(200)
+        .then(({body: {comments}}) => {
+            expect(comments).toBeInstanceOf(Array);
```
Comment (RichardNeat) [inline __tests__/app.test.js]: same with this array test

**Review by RichardNeat — COMMENTED** [review]
Nice work - just a couple of minor comments. Please update then commit again, then it's good to merge :)

**Pull request merged**